virttest/shared/deps/serial/serial_host_send_receive.py

```diff
@@ -222,6 +222,8 @@ def main():
     chardev_type = options.type
     if options.socket:
         device = options.socket
+        if '@' in device:
+            device = device.replace('@', '\0')
```

Comment:
It is better to use a separate commit and explain the reason for this change.

Reply:
> @nanliu-r We need further discussion on this, seems you didn't find the root cause of your issue.

@vivianQizhu update and please help to review again. Thanks.